### PR TITLE
Enable setting ACL 

### PR DESCRIPTION
**Pull request by phoban01**
This PR addresses #48.

### CODE REVIEW

**Comment by andris-sevcenko** [comment]
Have you tested this? It would seem to me that setting the ACL key on the config array is pointless, because the Flysystem adapter looks at the visibility key (https://github.com/thephpleague/flysystem-aws-s3-v3/blob/master/src/AwsS3Adapter.php#L615-L620).

A better approach would be to override the `visibility()` method (https://github.com/craftcms/cms/blob/develop/src/base/FlysystemVolume.php#L353-L356)

**Comment by phoban01** [comment]
@andris-sevcenko Ah yes, this makes a lot more sense 🐵 . (I had tested and it did work but overriding visibility is the way to go 👍)

File: src/Volume.php [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<?php
+// <?php
```
Comment (andris-sevcenko) [inline src/Volume.php]: Probably should not comment this one out.

File: src/Volume.php [hunk 2]
```diff
@@ -24,6 +24,8 @@
 use craft\helpers\StringHelper;
 use DateTime;
 use League\Flysystem\AwsS3v3\AwsS3Adapter;
+use League\Flysystem\AdapterInterface;
+
```
Comment (andris-sevcenko) [inline src/Volume.php]: no extra line

File: src/Volume.php [hunk 3]
```diff
@@ -106,6 +108,11 @@ public static function displayName(): string
      */
     public $expires = '';
 
+    /**
+     * @var string Set ACL for Uploads
+     */
+	public $acl = '';
```
Comment (andris-sevcenko) [inline src/Volume.php]: spaces, not tabs

File: src/Volume.php [hunk 4]
```diff
@@ -456,4 +463,17 @@ private static function _buildConfigArray($keyId = null, $secret = null, $region
...
+     *
+     * @return string
+     */
+	protected function visibility(): string {
+		if (!$this->acl) {
+			return $this->hasUrls ? AdapterInterface::VISIBILITY_PUBLIC : AdapterInterface::VISIBILITY_PRIVATE;
+		} else {
+			return $this->acl;
+		}
+	}
```
Comment (andris-sevcenko) [inline src/Volume.php]: also spaces not tabs.

File: src/templates/volumeSettings.html [hunk 5]
```diff
@@ -97,6 +97,18 @@
     placeholder: "path/to/subfolder"|t('aws-s3')
 }) }}
 
+{{ forms.autosuggestField({
+    label: "Upload ACL"|t('aws-s3'),
+	instructions: "Set the ACL for uploaded objects. If unset will default to value of the Public Urls field.",
```
Comment (andris-sevcenko) [inline src/templates/volumeSettings.html]: a rogue tab!

File: src/Volume.php [hunk 6]
```diff
@@ -456,4 +463,17 @@ private static function _buildConfigArray($keyId = null, $secret = null, $region
...
     }
+
+	 /**
+     * Returns the visibility setting for the Volume. If acl is set we ignore the hasUrls setting.
+     *
+     * @return string
+     */
+	protected function visibility(): string {
+		if (!$this->acl) {
+			return $this->hasUrls ? AdapterInterface::VISIBILITY_PUBLIC : AdapterInterface::VISIBILITY_PRIVATE;
```
Comment (andris-sevcenko) [inline src/Volume.php]: Better to call the parent method here instead of duplicating code.

**Comment by andris-sevcenko** [comment]
Thanks so much! I've set time aside tomorrow to test this. At a glance, this looks superb!

**Comment by phoban01** [comment]
Awesome, thanks for the guidance 😄

**Comment by andris-sevcenko** [comment]
Thanks for contributing!

**Pull request merged**